### DISCUSSION TITLE
EventAggregator versus Messenger

### DISCUSSION THREAD

**Discussion by YkTru** [PrismLibrary/Prism#2991] [General]
1- What would be the advantages/disavantages (flexibility, readability, maintainability, functionality etc.) between using Prism EventAggregator and https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/messenger? 

On first sight CommunityToolkit Messenger seems to me easier to implement, and less boilerplate. Is it a more "modern" approach to Prism EventAggregator?

2- A really nice feature of CommToolkit Messenger is "Request Messages":

```
*Using request messages*
" Another useful feature of messenger instances is that they can also be used to request values from a module to another. 
In order to do so, the package includes a base RequestMessage<T> : "


// Create a message
public class LoggedInUserRequestMessage : RequestMessage<User>
{
}

// Register the receiver in a module
WeakReferenceMessenger.Default.Register<MyViewModel, LoggedInUserRequestMessage>(this, (r, m) =>
{
    // Assume that "CurrentUser" is a private member in our viewmodel.
    // As before, we're accessing it through the recipient passed as
    // input to the handler, to avoid capturing "this" in the delegate.
    m.Reply(r.CurrentUser);
});

```

I can't find anything similar in Prism EventAggregator API, is it a deliberate design choice (is Request a bad practice?) or is it coming in Prism 9, or is there a way to "simulate"/implement this feature using Prism's EventAggregator?

Thanks

**Comment by sigmarsson** [comment]
Opt for `EventAggregator` unless this MVVM toolkit is capable to buffer messages. Prism is not - i.e. until the subscription is done, dispatched messages are lost - but it is okay.
